backend/src/extension.ts

```diff
@@ -135,34 +136,45 @@ export class YeomanUIPanel {
 						const resolve = this.questionsResolutions.get(message.taskId);
 						resolve(message.data);
 						return;
-					case 'showInfoMessage':
-						let InfoMessage = _.get(message, "commandParams[0]");
-						vscode.window.showInformationMessage(InfoMessage);
+					case 'showInstallMessage':
+						YeomanUIPanel.installing = true;
+						vscode.window.withProgress({
+							location: vscode.ProgressLocation.Notification,
+							title: "Installing dependencies..."
+						},
+						async (progress) => {
+							await new Promise(resolve => {
+								let intervalId = setInterval(() => {
+									if (!YeomanUIPanel.installing) {
+										clearInterval(intervalId);
+										resolve(undefined);
+									}
+								}, 3000);
+							});
+							return "installing_dependencies_completed";
+						});
```

Comment:
create 2 private methods from case 'showInstallMessage' and case 'showDoneMessage'.
It will help you to run the methods one after another and check the logic of both of them